rich/traceback.py

```diff
@@ -163,15 +163,28 @@ def ipy_display_traceback(
     try:  # pragma: no cover
         # if within ipython, use customized traceback
         ip = get_ipython()  # type: ignore[name-defined]
+        old_hooks = [ip._showtraceback, ip.showtraceback, ip.showsyntaxerror]
         ipy_excepthook_closure(ip)
-        return sys.excepthook
+        return old_hooks
```

Comment:
This method now returns two quite different return types, if I read this right.

Feels like a code smell. How about we return a callable which uninstalls?

Reply:
Hey! Thanks for having a look at this. It has been a while so I have forgotten some details. Let me have a look today and come back to this.